fideslog/api/schemas/analytics_event.py

```diff
@@ -83,14 +84,15 @@ class AnalyticsEvent(BaseModel):
         description="For events submitted as a result of making API server requests, the HTTP status code included in the response.",
     )
 
-    @validator("client_id")
-    def check_not_an_email_address(cls, value: str) -> str:
-        """
-        Validate that client_id does not contain an email address literal.
-        """
+    _check_not_an_email_address: classmethod = validator(
+        "client_id",
+        allow_reuse=True,
+    )(check_not_an_email_address)
 
-        assert value.find("@") == -1, "client_id must not be identifiable"
-        return value
+    _check_in_the_past: classmethod = validator(
```

Comment:
Just to check here - these timestamps are generated server-side right? I feel like I saw them in the client code as well, and while it's likely to work fine most of the time eventually a bad system clock could be an issue...

Reply:
Correct - these timestamps are created during the database interactions that create/update the `registrations` records. The validation needs to exist on the off chance that a user tries to populate these fields manually.